clarity-ext-scripts/clarity_ext_scripts/covid/fetch_biobank_barcodes.py

```diff
@@ -0,0 +1,137 @@
+import re
+from clarity_ext.service.file_service import Csv
+from clarity_ext.domain.validation import UsageError
+from clarity_ext.utils import single
+
+BIOBANK_FILE_HEADER = ['well', 'biobank_barcode', 'something_else', 'plate_barcode']
+
+
+class FetchBiobankBarcodes(object):
+    """
+    This is intended to be part of create samples script. By injecting context,
+    it might be tested in isolation by creating a dummy script.
+    """
+
+    def __init__(self, context):
+        self.context = context
+        self.barcode_by_sample_code = None
+
+    def execute(self):
+        """
+        Just for testing, validate() and map_barcodes() are called separately from
+        different scripts
+        """
+        self.validate()
+        self.barcode_by_sample_code =\
+            self.biobank_barcode_by_sample_referal_code()
+        self._print(self.barcode_by_sample_code)
+
+    def validate(self):
+        try:
+            file_stream = self.context.local_shared_file('Raw biobank list')
+        except IOError:
+            raise UsageError("Please upload the file to 'Raw barcode list' before proceeding!")
+
+        biobank_matrix = self._build_biobank_matrix(file_stream)
+        plate_barcodes = {
+            biobank_matrix[key]['plate_barcode'] for key in biobank_matrix
+        }
+        if len(plate_barcodes) > 1:
+            raise UsageError("There are more than one destination plates the file in 'Raw barcode list'!")
+
+        # Validate that sample list file has plate barcode as name
+        filenames = self.context.file_service.list_filenames('Raw sample list')
+        base_names = [n.split('.')[0] for n in filenames]
+        plate_barcode = single(list(plate_barcodes))
+        if plate_barcode not in base_names:
+            raise UsageError(
+                "The 'Raw sample list' name is not matching with the plate "
+                "barcode in 'Raw biobank list', {}".format(plate_barcode))
+
+        # Validate that all wells in biobank file are matched in sample list file
+        file_stream2 = self.context.local_shared_file('Raw sample list')
+        sample_matrix = self._build_sample_matrix(file_stream2, plate_barcode)
+        for key in biobank_matrix:
```

Comment:
If the matrix would have the suggested name `biobank_info_by_well_barcode`, I would directly understand which kind of key to expect.